### ISSUE TITLE
Detect when Dropdown is shown in DOM

### ISSUE THREAD

**Issue by brendensoares** [ng-bootstrap/ng-bootstrap#1521]
Feature Request
---
The Dropdown has the [`openChanged`](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/src/dropdown/dropdown.ts#L40) event emitter that notifies observers when the dropdown is opened. Unfortunately, when the event is fired, the dropdown DOM element has [not yet been shown in the DOM](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/src/dropdown/dropdown.ts#L13).

Is there an easy way to ***notify observers when the dropdown is open and ready for DOM interactions***?

Possible Solutions
---
In order to (hopefully) wait for the DOM to be painted by the browser and thus the content to be present, I've considered zero delays (`setTimeout(cb, 0)`) as well as waiting for the next browser repaint ([`window.requestAnimationFrame()`](https://swizec.com/blog/how-to-properly-wait-for-dom-elements-to-show-up-in-modern-browsers/swizec/6663)). This seems hacky and unreliable.

The only other solution is to write an Angular 2 native component that shows/hides the dropdown based on `ngIf` instead of a CSS class.

The current implementation is very lightweight, but seems a bit lacking as far as extensibility.

Background
---
I'm trying to add an autofocusing `input`to the dropdown so that users can filter the list of options. I need to manually call `focus()` on the element when the dropdown is opened.

**Comment by pkozlowski-opensource** [comment]
@brendensoares not sure I can follow what you are trying to do. Please put together a plunker with a minimal reproduction of your use-case: https://github.com/ng-bootstrap/ng-bootstrap/blob/master/README.md#you-think-youve-found-a-bug

The thing is that we are not adding / removing DOM nodes when it comes to dropdown, just swapping CSS classes. So all dropdown elements should be there all the time.

In any case, we need more details here as it is not clear what you are trying to do / what is not working.

**Comment by brendensoares** [comment]
@pkozlowski-opensource it's not a bug, it's a change in how the dropdown toggles the dropdown list content.

It may be in the DOM, but since it is hidden with CSS, JS's `focus()` method does not focus the field. My request, as detailed above, is the possibility of managing the toggling of the list content through Angular's `ngIf` instead of CSS classes so that my app JS code can properly focus the input field within the dropdown list content.

Let me know if you need more clarification. Thanks.

**Comment by pkozlowski-opensource** [comment]
> Let me know if you need more clarification

Yeh, sorry, definitively more clarification would be needed :-) 

> but since it is hidden with CSS, JS's focus() method does not focus the field

Could you please try to reproduce it in a plunker? It is super hard for me to reason about things based on text-description only. Having a working code would be much better.

> possibility of managing the toggling of the list content through Angular's ngIf instead of CSS classes so that my app JS code can properly focus the input field within the dropdown list content.

This would mean divergence from the Bootstrap's way of doing things. I'm not saying that we can't do this but rather that we should have really good reasons before we do so. 

tl;dr; is that you should start a plunker going if you want to get it further.

**Comment by pkozlowski-opensource** [comment]
Closing since it is not clear from the description what is needed and we haven't heard back with a plunker.
